File: Backend/src/main/java/com/kas/online_book_shop/service/PostService.java
package com.kas.online_book_shop.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.kas.online_book_shop.enums.PostState;
import com.kas.online_book_shop.model.Post;
import com.kas.online_book_shop.model.PostCategory;

public interface PostService {

    Page<Post> getPostByTileContainingAndCategoryAndState(String title, PostCategory category, PostState state, Pageable pageable);

    Post savePost(Post post);

    Post updatePost(Post post);

    void deletePost(Long id);

    Post getPostById(Long id);

    List<Post> getAll();
}

File: Frontend/re-admin/src/service/PostService.js
import axios from "axios";

const API_URL = "http://localhost:8081/api/v1/post";

const getAllPosts = () => {
    return axios.get(API_URL);
}

const getPostById = (id) => {
    return axios.get(API_URL + '/' + id);
}

const createPost = (data) => {
    return axios.post(API_URL, data);
}

const updatePost = (data) => {
    return axios.put(API_URL, data);
}

const deletePost = (id) => {
    return axios.delete(API_URL + '/' + id);
}

const getAllPostCategories = () => {
    return axios.get("http://localhost:8081/api/v1/post-category");
}

export { getAllPosts, getPostById, createPost, updatePost, deletePost, getAllPostCategories}